### PR TITLE
Object and ObjectSet equality.

### PR DESCRIPTION
**Pull request by allenap**
Implement equality (`__eq__`) for `Object` and `ObjectSet`.

Also fix Device.read and test it, aided by this equality checking.

### CODE REVIEW

File: maas/client/utils/auth.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def obtain_token(url, username, password, *, insecure=False):
 
         # Extract the CSRF token.
         login_doc = bs4.BeautifulSoup(response.content, "html.parser")
-        login_button = login_doc.find('input', value="Login")
+        login_button = login_doc.find('button', text="Login")
```
Comment (blakerouse) [inline maas/client/utils/auth.py]: Is this going to work across 2.1 and 2.2?

Comment (allenap) [inline maas/client/utils/auth.py]: No, but this needs to be rewritten in any case. This just unblocks anyone working with MAAS trunk, like me :)

Comment (allenap) [inline maas/client/utils/auth.py]: In other words, I do intend to fix this, but not in this branch.

**Comment by allenap** [comment]
Okay, I don't seem to get Git or GitHub. I wanted to create a pull request with something like Launchpad's prerequisite branch, but it turns out I'd created a pull request to merge into client-connect-and-login.

**Pull request merged**